### PR TITLE
Finish support of the new gateway directory API

### PR DESCRIPTION
**Pull request by octol**
Finish support of the new gateway directory API on nymvpn.com.
Support setting the mixnet and vpn performance threshold parameters for the gateway directory.
Update grpc API.

The changes to the uniffi API is split out into https://github.com/nymtech/nym-vpn-client/pull/1155

**NOTE**: breaks the grpc API for getting gateways and countries

### CODE REVIEW

File: proto/nym/vpn.proto [hunk 1]
```diff
@@ -353,78 +354,55 @@ message AsExit {
...
+message ListGatewaysResponse {
+  repeated GatewayResponse gateways = 1;
 }
 
-message ListVpnCountriesRequest {
-  Threshold min_gateway_performance = 1;
+message ListCountriesRequest {
+  GatewayType kind = 1;
+  Threshold min_mixnet_performance = 2;
+  Threshold min_vpn_performance = 3;
```
Comment (doums) [inline proto/nym/vpn.proto:402]: These two fields should be both marked as `optional`

File: proto/nym/vpn.proto [hunk 2]
```diff
@@ -353,78 +354,55 @@ message AsExit {
...
+  EXIT = 2;
+  VPN = 3;
 }
 
-message ListExitGatewaysResponse {
-  repeated ExitGateway gateways = 1;
+message ListGatewaysRequest {
+  GatewayType kind = 1;
+  Threshold min_mixnet_performance = 2;
+  Threshold min_vpn_performance = 3;
```
Comment (doums) [inline proto/nym/vpn.proto:392]: These two fields should be both marked as `optional`

File: proto/nym/vpn.proto (hunk 1 above)
Comment (doums) [reply proto/nym/vpn.proto:405]: It is an `Option` in rust implementations yes, but is it the case with other languages' implementation, like in swift?

File: proto/nym/vpn.proto [hunk 3]
```diff
@@ -353,78 +355,57 @@ message AsExit {
...
   Probe last_probe = 3;
 }
 
-message ListExitGatewaysRequest {
-  Threshold min_gateway_performance = 1;
+enum GatewayType {
+  GATEWAY_TYPE_UNSPECIFIED = 0;
+  ENTRY = 1;
+  EXIT = 2;
+  VPN = 3;
```
Comment (doums) [inline proto/nym/vpn.proto:387]: Suggestion:

```
MIXNET_ENTRY
MIXNET_EXIT
WG
```

**Review by doums — APPROVED** [review]

File: proto/nym/vpn.proto (hunk 3 above)
Comment (octol) [reply proto/nym/vpn.proto:387]: yes good suggestion

**Comment by doums** [comment]
Changed the proto to match

```
MIXNET_ENTRY
MIXNET_EXIT
WG
```

**Review by zaneschepke — APPROVED** [review]

File: proto/nym/vpn.proto (hunk 1 above)
Comment (doums) [reply proto/nym/vpn.proto:405]: @rokas-ambrazevicius are proto non-scalar types generated in Swift code as "optional" or "nullable" without being labeled as `optional`?

**Pull request merged**